livekit-rtc/livekit/rtc/frame_processor.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar, Union
+from .audio_frame import AudioFrame
+from .video_frame import VideoFrame
+
+
+T = TypeVar("T", bound=Union[AudioFrame, VideoFrame])
+
+
+class FrameProcessor(Generic[T], ABC):
+    @property
+    @abstractmethod
+    def is_enabled(self) -> bool: ...
+
+    @abstractmethod
+    def set_enabled(self, enable: bool): ...
+
+    @abstractmethod
+    def _update_stream_info(
+        self,
+        *,
+        room_name: str,
+        participant_identity: str,
+        publication_sid: str,
+    ): ...
+
+    @abstractmethod
+    def _update_credentials(self, *, token: str, url: str): ...
```

Comment:
nit: If this is called by an external event, maybe we can call those events like this? 
So it is obvious for the users implementing them that this is external. 
```
def on_stream_info_updated()

def on_credentials_updated()
```